### DISCUSSION TITLE
Overwrite confirmation

### DISCUSSION THREAD

**Discussion by edgarogh** [diesel-rs/diesel#2690] [Ideas]
The diesel cli just overwrote a big uncommited `src/schema.rs` file in my codebase that contained graphql-related stuff (nothing to do with the database). It would've been very useful to have some kind of confirmation prompt if diesel detects that the file isn't formatted like its usual `schema.rs` files with the macro call as a first statement. This would prevent other data losses.

**edit:** fortunately, IntelliJ/CLion keeps a little revision history for every file and I was able to recover its content

**Comment by weiznich** [comment]
To state that very clearly here: Diesel cli does not just override random files, but only that file that is explicitly configured via CLI arguments or the `diesel.toml` file. 
That written, we would likely accept a PR adding another check here to warn about modified `schema.rs` files. Such a PR should include at least a few test demonstrating the functionality. At least for me personally this is nothing that is on my todo list for the next time.

**Reply by edgarogh** [reply]
I'm aware that the CLI only writes to the configured file and I should have remembered that it defaulted to `src/schema.rs`, I'm completely at fault and wasn't trying to imply otherwise, I'm really sorry if my intentions weren't clear. I was suggesting this as a little addition to prevent other people from losing data like I could have done.

I might work on this PR if I have a little bit of free time, I understand that this is a very minor problem. Thank you for your answer, and thank you for your work on diesel. It's the second, and certainly not the last time I'm using it !
